src/biokbase/narrative/jobs/job.py

```diff
@@ -239,19 +240,10 @@ def __getattr__(self, name):
         return attr[name]()
 
     def __setattr__(self, name, value):
-        if name in STATE_ATTRS:
-            self._acc_state[name] = value
-        elif name in JOB_INPUT_ATTRS:
-            self._acc_state["job_input"] = self._acc_state.get("job_input", {})
-            self._acc_state["job_input"][name] = value
-        elif name in NARR_CELL_INFO_ATTRS:
-            self._acc_state["job_input"] = self._acc_state.get("job_input", {})
-            self._acc_state["job_input"]["narrative_cell_info"] = self._acc_state[
-                "job_input"
-            ].get("narrative_cell_info", {})
-            self._acc_state["job_input"]["narrative_cell_info"][name] = value
-        else:
-            object.__setattr__(self, name, value)
+        if name in ALL_ATTRS:
+            raise AttributeError("Job attributes must be updated using the `update_state` method")
+
+        object.__setattr__(self, name, value)
```

Comment:
anything that's not a job attribute can use the standard attribute setter

Reply:
You have to be careful not to set anything that's *just* a lone nested field though (i.e. stuff that comes in as job input) because that would overwrite everything in those nested fields

Reply:
Yep -- the only potential place that could occur was in the `parameters()` method, which I rejigged.